File: anplayer-plugin/src/main/java/com/xeinebiu/anplayer/plugin/base/media/extractor/extractor/SearchExtractor.kt
package com.xeinebiu.anplayer.plugin.base.media.extractor.extractor

import com.xeinebiu.anplayer.plugin.base.media.extractor.model.descriptor.MediaDescriptor

/**
 * Extractor to handle queries
 */
abstract class SearchExtractor : Extractor<MediaDescriptor>() {

    /**
     * Returns current query
     */
    abstract fun getQuery(): String?
}

File: packages/plugin/src/lib/extractor/search.extractor.ts
import { Extractor } from './extractor';
import {
    Album,
    AlbumDetail,
    Author,
    AuthorDetail,
    Media,
    MediaDetail,
} from '../model';

export type SearchExtractor = Extractor<
    Media | Album | Author | MediaDetail | AlbumDetail | AuthorDetail,
    { getQuery(): string }
>;
